### PR TITLE
Integrate Redux in Settings, Remove Flux

### LINKED ISSUE #2127: Integrate Redux in Settings, Remove Flux integration in settings
**Actual Behaviour**

Currently not present

**Expected Behaviour**

Integrate redux in settings

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by fragm3**
Fixes #2127

Changes:

1. Integrate Redux
2. Change Architecture Of Settings Component #2130
3. Remove flux and all its related files #2133 
4. Remove UserPreferenceStore and UserIdentityStore #2132
5. Fix bug in Settings #2131
6. Added Snackbar, Loader
7. Added test
8. Migrate some component to styled-components.

Demo Link: https://pr-2128-1-fossasia-susi-web-chat.surge.sh

### CODE REVIEW

**Comment by fragm3** [comment]
Custom theme changer will work after PR https://github.com/fossasia/susi_server/pull/1264 being merged.

**Comment by fragm3** [comment]
Custom theme changer will work now.

File: src/components/ChatApp/Settings/AccountTab.react.js [hunk 1]
```diff
@@ -1,98 +1,263 @@
...
+              userName === this.props.userName) ||
+            userNameError
+          }
+          style={{ marginTop: '1.5rem' }}
+        >
+          <Translate text="Save Changes" />
+        </Button>
+        <div style={{ marginRight: '20px' }}>
+          {theme === 'light' ? (
+            <hr
```
Comment (akshatnitd) [inline src/components/ChatApp/Settings/AccountTab.react.js]: Make this into a styled component.

File: src/components/ChatApp/Settings/Settings.react.js [hunk 2]
```diff
@@ -108,826 +85,57 @@ class Settings extends Component {
...
-
-  // eslint-disable-next-line
   componentDidMount() {
-    document.body.className = 'white-body';
-    if (!this.state.dataFetched && cookies.get('loggedIn')) {
-      this.apiCall();
+    const { accessToken, actions } = this.props;
+    if (accessToken) {
+      actions.getUserSettings().then(() => {
+        this.setState({ loading: false, theme: this.props.theme });
```
Comment (akshatnitd) [inline src/components/ChatApp/Settings/Settings.react.js]: Does this.props.theme return the updated theme? Should we take the payload from the server API response in then block and then update it?

File: src/components/ChatApp/Settings/ShareOnSocialMedia.js [hunk 3]
```diff
@@ -84,13 +85,14 @@ const ShareOnSocialMedia = props => {
   initFacebookAPI();
   const { raisedButtonStyle } = styles;
   const { classes } = props;
+  console.log('TEST');
```
Comment (akshatnitd) [inline src/components/ChatApp/Settings/ShareOnSocialMedia.js]: Remove it

File: src/redux/reducers/settings.js [hunk 4]
```diff
@@ -33,89 +30,83 @@ const defaultState = {
   serverUrl: 'https://api.susi.ai',
   backgroundImage: '',
   messageBackgroundImage: '',
+  avatarType: 'gravatar',
```
Comment (akshatnitd) [inline src/redux/reducers/settings.js]: Set the default avatarType to 'default'

File: src/redux/reducers/settings.js [hunk 5]
```diff
@@ -33,89 +30,83 @@ const defaultState = {
...
-    [actionTypes.SETTINGS_SET_CHAT_PREFERENCES_SETTINGS](state, { payload }) {
-      const { enterAsSend } = payload;
+        phoneNo,
+        checked,
+        backgroundImage,
+        messageBackgroundImage,
+        avatarType,
+      } = settings;
+      let { customThemeValue } = settings;
+      const themeArray = customThemeValue.split(',').map(value => `#${value}`);
```
Comment (akshatnitd) [inline src/redux/reducers/settings.js]: What if the use uses light or dark theme instead of custom?

Comment (fragm3) [reply src/redux/reducers/settings.js]: @akshatnitd If we use light or dark theme, the values will be fixed for both of them.

**Comment by fragm3** [comment]
> The custom theme values is applied even if I have selected dark or light theme. Please check this once. 

@akshatnitd Should I make the changes in this PR itself? or create a new PR after this is merged?

File: src/components/ChatApp/Settings/Settings.react.js (hunk 2 above)
Comment (fragm3) [reply src/components/ChatApp/Settings/Settings.react.js]: @akshatnitd Yes, API returns theme. It is necessary to update the theme, I could I get it from the response or from props(store), similar stuff, right?

**Comment by fragm3** [comment]
@akshatnitd I am almost done with the changes will rebase soon.

**Comment by fragm3** [comment]
@akshatnitd made the changes 😃

File: src/components/ChatApp/Settings/Settings.react.js (hunk 2 above)
Comment (akshatnitd) [reply src/components/ChatApp/Settings/Settings.react.js]: Yes, it is similar. But I was checking, if you get the updated props in the then block or not.

File: src/components/ChatApp/MessageSection/MessageSection.react.js [hunk 6]
```diff
@@ -431,35 +448,34 @@ class MessageSection extends Component {
...
+    let header, pane, body, composer, button;
+
+    if (theme === 'custom') {
+      header = customThemeValue.header;
+      pane = customThemeValue.pane;
+      body = customThemeValue.body;
+      composer = customThemeValue.composer;
+      button = customThemeValue.button;
+    } else {
+      header = theme === 'dark' ? dark.header : light.header;
```
Comment (akshatnitd) [inline src/components/ChatApp/MessageSection/MessageSection.react.js]: The logic is too complex for it and involves too many conditions across files. Instead set the const object of dark and light theme in `utils/colors` it to customTheme value at the reducer itself.

Comment (fragm3) [reply src/components/ChatApp/MessageSection/MessageSection.react.js]: On it

File: src/components/ChatApp/Settings/Settings.react.js (hunk 2 above)
Comment (fragm3) [reply src/components/ChatApp/Settings/Settings.react.js]: Tried it, it's behaving weirdly, loader keeps loading. Don't know why @akshatnitd

**Comment by fragm3** [comment]
@akshatnitd made the changes.

**Pull request merged**